velox/common/memory/SharedArbitrator.cpp

```diff
@@ -803,10 +816,9 @@ void SharedArbitrator::globalArbitrationMain() {
   while (true) {
     {
       std::unique_lock l(stateLock_);
-      globalArbitrationThreadCv_.wait(l, [&] {
-        return globalArbitrationStop_ || !globalArbitrationWaiters_.empty();
-      });
-      if (globalArbitrationStop_) {
+      globalArbitrationThreadCv_.wait(
+          l, [&] { return shutdown_ || !globalArbitrationWaiters_.empty(); });
```

Comment:
locking on stateLock_ here waiting for shutdown_ to change but changing shutdown_ needs stateLock_, too. Will we run into deadlock?